src/util/templates/ChainedTemplateEngine.ts

```diff
@@ -27,12 +29,10 @@ export class ChainedTemplateEngine<T extends Dict<any> = Dict<any>> implements T
     this.renderedName = renderedName;
   }
 
-  public async render(contents: T): Promise<string>;
-  public async render<TCustom = T>(contents: TCustom, template: Template): Promise<string>;
-  public async render<TCustom = T>(contents: TCustom, template?: Template): Promise<string> {
-    let body = await this.firstEngine.render(contents, template!);
+  public async handle({ contents, template }: TemplateEngineInput<T>): Promise<string> {
+    let body = await this.firstEngine.handleSafe({ contents, template });
```

Comment:
We can't fully check since the data is dynamic, but this class could now have a `canHandle` call that checks `this.firstEngine.canHandle`, and then only calls `this.firstEngine.handle` here since we already checked.